smithy-model/src/main/java/software/amazon/smithy/model/shapes/ModelSerializer.java

```diff
@@ -59,7 +59,8 @@ private ModelSerializer(Builder builder) {
         } else {
             shapeFilter = builder.shapeFilter;
         }
-        traitFilter = builder.traitFilter;
+        // Never serialize ephemeral traits.
+        traitFilter = builder.traitFilter.and(FunctionalUtils.not(Trait::isEphemeral));
```

Comment:
Wouldn't it be kind of dangerous to serialize a model that has these traits at all? It would no longer be usable. maybe we should throw an error rather than filtering the traits.

Reply:
No, the model is definitely usable without ephemeral traits. The point of these traits is that they’re only useful for intermediate transforms. For example, you might want to punch traits on to shapes only for a specific code generator, but the traits don’t need to be persisted. Every projection in Smithy is serialized, so I don’t want ephemeral traits to poison models.